src/functional/ffront/func_to_foast.py

```diff
@@ -42,6 +42,44 @@ class FieldOperatorSyntaxError(DialectSyntaxError):
     dialect_name = "Field Operator"
 
 
+class UnrollPowerOp(eve.NodeTranslator):
+    def visit_BinOp(self, node: foast.BinOp) -> foast.BinOp:
+        if node.op == foast.BinaryOperator.POW:
+            new_left = self.visit(node.left)
+            if int(node.right.value) == 1:
+                return self.visit(node.left)
+            if int(node.right.value) == 0:
+                return foast.BinOp(
+                    left=new_left,
+                    right=new_left,
+                    op=foast.BinaryOperator.DIV,
+                    location=node.location,
+                    type=node.type,
+                )
+
+            new_op = foast.BinaryOperator.MULT
+            new_left_BinOp = foast.BinOp(
+                left=new_left, right=new_left, op=new_op, location=node.location, type=node.type
+            )
+            for _i in range(int(float(node.right.value) - 2)):
+                new_left_BinOp = foast.BinOp(
+                    left=new_left_BinOp,
+                    right=new_left,
+                    op=new_op,
+                    location=node.location,
+                    type=node.type,
+                )
+            return new_left_BinOp
```

Comment:
```suggestion
            new_left = self.visit(node.left)
            if int(node.right.value) == 0:
                return foast.BinOp(
                    left=new_left,
                    right=new_left,
                    op=foast.BinaryOperator.DIV,
                    location=node.location,
                    type=node.type,
                )

            unrolled_expr = self.visit(node.left)
            for _i in range(int(node.right.value) - 1):
                unrolled_expr = foast.BinOp(
                    left=unrolled_expr,
                    right=new_left,
                    op=foast.BinaryOperator.MULT,
                    location=node.location,
                    type=node.type,
                )
            return unrolled_expr
```
Almost there. You can further get rid of the special casing for an exponent of one.